src/test/java/core/basesyntax/dao/RegistrationServiceTest.java

```diff
@@ -0,0 +1,113 @@
+package core.basesyntax.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationValidationException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void clearStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_ValidUser_Ok() {
+        User user = new User("testlogin", "password123", 25);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals("testlogin", registeredUser.getLogin());
+        assertEquals("password123", registeredUser.getPassword());
+        assertEquals(25, registeredUser.getAge());
+        assertNotNull(registeredUser.getId());
+    }
+
+    @Test
+    void register_loginShorterThanRequired_notOk() {
+        User user = new User("log", "password123", 25);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user);
+        }, "RegistrationValidationException "
+                + "should be thrown if the login has less than 6 characters");
+    }
+
+    @Test
+    void register_passwordShorterThanRequired_notOk() {
+        User user = new User("testlogin", "pass", 25);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user);
+        }, "RegistrationValidationException "
+                + "should be thrown if the password has less than 6 characters");
+    }
+
+    @Test
+    void register_ageLessThanRequired_notOk() {
+        User user = new User("testlogin", "password", 12);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user);
+        }, "RegistrationValidationException "
+                + "should be thrown if age is less than 18");
+
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        User user1 = new User("testlogin", "password", 25);
+        Storage.people.add(user1);
+
+        User user2 = new User("testlogin", "password123", 30);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user2);
+        }, "RegistrationValidationException "
+                + "should be thrown if the login already exists");

```

Comment:
This test checks for duplicate logins, which is good. However, ensure that the `RegistrationServiceImpl` contains logic to actually check for duplicates. If it doesn't, this test will fail for the wrong reason.